src/serializer/ResidualHeapSerializer.js

```diff
@@ -1749,6 +1748,8 @@ export class ResidualHeapSerializer {
     // Make sure additional functions get serialized.
     let rewrittenAdditionalFunctions = this.processAdditionalFunctionValues();
 
+    Array.prototype.push.apply(this.prelude, this.preludeGenerator.prelude);
```

Comment:
If `processAdditionalFunctionValues` (indirectly) causes the `preludeGenerator` to get some new entries, then this is clearly the right thing to do.

Adding something to prevent any further modifications to the PreludeGenerator after this point would be nice, but can be dome later separately.

Reply:
I feel like it's probably not processAdditionalFunctionValues that adds to the generator, its probably [this chunk of the visitor](https://github.com/facebook/prepack/blob/master/src/serializer/ResidualHeapVisitor.js#L918-L930)